src/Compilers/CSharp/Portable/Symbols/Source/SourceComplexParameterSymbol.cs

```diff
@@ -1048,6 +1057,199 @@ bool needsReporting()
             }
         }
 
+#nullable enable
+        private void DecodeInterpolatedStringHandlerArgumentAttribute(ref DecodeWellKnownAttributeArguments<AttributeSyntax, CSharpAttributeData, AttributeLocation> arguments, Binder binder, BindingDiagnosticBag diagnostics)
```

Comment:
I guess i'm suprised that we would need the binder in order to decode the attribute. I see below that we're doing overload resolution to determine if its valid or not. Would an alternative be to just decode the raw strings here? Then wherever we actually use them in the binder, do the actual resolution etc? We can always get back to them from the containing symbol I think? (they always apply to other parameters in the same list as the one the attribute is on?)

Reply:
> Would an alternative be to just decode the raw strings here?

We could, yes, but then you don't get validation of the attribute at the attribute definition location. This seems suboptimal to me.

Reply:
Hmm, yeah thats a bummer. Just seems like we're doing a lot of work here that as a caller I woulnd't expect to happen. Do we have anywhere else where the attribute itself can error based on semantics of other code?

---
In reply to: [647835991](https://github.com/dotnet/roslyn/pull/53932#discussion_r647835991) [](ancestors = 647835991)

Reply:
Hmm. Could we not just store the location along with the cracked data? So we can report on it if it fails later on?

Reply:
> So we can report on it if it fails later on?

This doesn't help if you never use it in this project (as I would expect many to do, since these will mostly be used in libraries). Unless Aleksey has a different opinion or suggestion for how to approach this, I believe this is the right way to go.

Reply:
After discussion with Aleksey, we've decided that there's too much potential for issue with missing references, retargeting, and langversion changes to move forward with this for now. I'll remove the binder code.